### DISCUSSION TITLE
How can i build libmetal for a barametal soc?

### DISCUSSION THREAD

**Discussion by kikuyuchanj** [OpenAMP/libmetal#310] [Q&A]
I want to build libmetal and openamp for a barametal system, which is not Xillinx. But when I build cmake with newlib toolchain, I meet this problem:
-- Found Threads: TRUE
CMake Error at /usr/share/cmake-3.10/Modules/FindPackageHandleStandardArgs.cmake:137 (message):
  Could NOT find LibRt (missing: LIBRT_LIBRARIES)
Call Stack (most recent call first):
  /usr/share/cmake-3.10/Modules/FindPackageHandleStandardArgs.cmake:378 (_FPHSA_FAILURE_MESSAGE)
  cmake/modules/FindLibRt.cmake:36 (find_package_handle_standard_args)
  cmake/depends.cmake:24 (find_package)
  CMakeLists.txt:24 (include)

**Comment by arnopo** [comment]
your issue seems linked to your environment.  
The issue is reported  [here](https://github.com/OpenAMP/libmetal/blob/main/cmake/depends.cmake#L24), indicating that you are building the libraries for a Linux target, not a baremetal target
